src/backend/cdb/cdbpath.c

```diff
@@ -365,6 +377,12 @@ cdbpath_create_motion_path(PlannerInfo *root,
 	 * materialize nodes on top of motion nodes
 	 */
 
+	/*
+	 * If the subpath requires parameters, we cannot generate Motion atop of it.
+	 */
+	if (!bms_is_empty(PATH_REQ_OUTER(subpath)))
+		return NULL;
```

Comment:
@maksm90 I was confused about the type of subpath varaible. It's the argument passed by reference. So in case of condition is true and there is Materialize on top of subpath, we'll remove it anyway after patch.